### ISSUE TITLE
Invalid ouput

### ISSUE THREAD

**Issue by monolithed** [bcherny/json-schema-to-typescript#66]
```ts
import * as jsonScheme from 'generate-schema';
import { compile } from 'json-schema-to-typescript';

let scheme = jsonScheme.json('x', {
	"x[0]": 1
});

let result = compile(scheme, undefined);

console.log(result)
```

```ts
export interface X {
      x[0]?: number; 
      [k: string]: any;
}
```

`x[0]` is not a valid name. What is expected:

```ts
export interface X {
      'x[0]'?: number; 
      [k: string]: any;
}
```

**Comment by bcherny** [comment]
What is `scheme` - can you log it out?

**Comment by monolithed** [comment]
> What is scheme - can you log it out?

```js
{ '$schema': 'http://json-schema.org/draft-04/schema#',
  title: 'x',
  type: 'object',
  properties: { 'x[0]': { type: 'number' } } }
```

**Comment by bcherny** [comment]
PRs are welcome - the bug is probably on [this line](https://github.com/bcherny/json-schema-to-typescript/blob/83f847594278c386d49c3c75f8052d7c0c9bad9f/src/TsTypes.ts#L165), and we should escape strings containing special characters.

**Comment by bcherny** [comment]
Published 3.1.4
